examples/torch/pearl_ml1_push.py

```diff
@@ -0,0 +1,163 @@
+    net_size=300,
+    meta_batch_size=16,
+    num_steps_per_epoch=4000,
+    num_initial_steps=4000,
+    num_tasks_sample=15,
+    num_steps_prior=750,
+    num_extra_rl_steps_posterior=750,
+    num_steps_per_eval=450,
+    batch_size=256,
+    embedding_batch_size=64,
+    embedding_mini_batch_size=64,
+    max_path_length=150,
+    reward_scale=10.,
+    use_information_bottleneck=True,
+    use_next_obs_in_context=False,
+    use_gpu=True,
+)
+
+
+def run_task(snapshot_config, *_):
+    """Set up environment and algorithm and run the task.
+
+    Args:
+        snapshot_config (garage.experiment.SnapshotConfig): The snapshot
+            configuration used by LocalRunner to create the snapshotter.
+            If None, it will create one with default settings.
+        _ : Unused parameters
+
+    """
+    # create multi-task environment and sample tasks
+    env_sampler = SetTaskSampler(lambda: GarageEnv(
+        normalize(ML1.get_train_tasks('push-v1'))))
+    env = env_sampler.sample(params['num_train_tasks'])
+
+    test_env_sampler = SetTaskSampler(lambda: GarageEnv(
+        normalize(ML1.get_test_tasks('push-v1'))))
+    #test_env = test_env_sampler.sample(params['num_train_tasks'])
+
+    runner = LocalRunner(snapshot_config)
+    obs_dim = int(np.prod(env[0]().observation_space.shape))
+    action_dim = int(np.prod(env[0]().action_space.shape))
+    reward_dim = 1
+
+    # instantiate networks
+    encoder_in_dim = obs_dim + action_dim + reward_dim
+    encoder_out_dim = params['latent_size'] * 2
+    net_size = params['net_size']
+
+    context_encoder = MLPEncoder(input_dim=encoder_in_dim,
+                                 output_dim=encoder_out_dim,
+                                 hidden_sizes=[200, 200, 200])
+
+    space_a = akro.Box(low=-1,
+                       high=1,
+                       shape=(obs_dim + params['latent_size'], ),
+                       dtype=np.float32)
+    space_b = akro.Box(low=-1, high=1, shape=(action_dim, ), dtype=np.float32)
+    augmented_env = EnvSpec(space_a, space_b)
+
+    qf1 = ContinuousMLPQFunction(env_spec=augmented_env,
+                                 hidden_sizes=[net_size, net_size, net_size])
+
+    qf2 = ContinuousMLPQFunction(env_spec=augmented_env,
+                                 hidden_sizes=[net_size, net_size, net_size])
+
+    obs_space = akro.Box(low=-1, high=1, shape=(obs_dim, ), dtype=np.float32)
+    action_space = akro.Box(low=-1,
+                            high=1,
+                            shape=(params['latent_size'], ),
+                            dtype=np.float32)
+    vf_env = EnvSpec(obs_space, action_space)
+
+    vf = ContinuousMLPQFunction(env_spec=vf_env,
+                                hidden_sizes=[net_size, net_size, net_size])
+
+    policy = TanhGaussianMLPPolicy(env_spec=augmented_env,
+                                   hidden_sizes=[net_size, net_size, net_size])
+
+    context_conditioned_policy = ContextConditionedPolicy(
```

Comment:
since PEARL doesn't support anything but this policy, I think it would be more natural to pass the policy and encoder and have PEARL internally construct a ContextConditionedPolicy. That would also hide a lot of this nasty space munging from the launcher

Reply:
Yeah, that would probably be better.